lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -549,5 +549,42 @@ sub export {
     $self->finish('END');
 }
 
+sub module_fails {
+    my ($self) = @_;
+
+    return $self->reply->not_found if (!defined $self->param('testid') or !defined $self->param('moduleid'));
```

Comment:
no post-condition if the line becomes longer than 30 characters please.

And in this case unless (A and B) is actually easier to read than if (!A or !B)